### PR TITLE
[BEAM-5879 ] Make write_record() in tfrecordio.py py3 compatible

### PR DESCRIPTION
**Pull request by ruoyu90**
Make write_record() in tfrecordio.py py3 compatible

- [BEAM-5879 ]

### CODE REVIEW

**Comment by ruoyu90** [comment]
@tvalentyn

File: sdks/python/apache_beam/io/tfrecordio.py [hunk 1]
```diff
@@ -78,7 +78,10 @@ def _masked_crc32c(cls, value, crc32c_fn=_default_crc32c_fn):
       Masked crc32c checksum.
     """
 
-    crc = crc32c_fn(value)
+    if isinstance(value, bytes):
```
Comment (tvalentyn) [inline sdks/python/apache_beam/io/tfrecordio.py]: Would anything break if we always follow the `else` branch? If yes, why? 
Also, last letter in `crc32c_fu` is a typo?

**Comment by tvalentyn** [comment]
cc: @markflyhigh

File: sdks/python/apache_beam/io/tfrecordio.py (hunk 1 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/io/tfrecordio.py]: Before this change, the default codepath was:
`crc = crc32c_fn(value)`
I assume this did not work on Python 3, so, I suspected that on Python 3 `value` is a `str`. Are you saying that `value` is sometimes `str` and sometimes `bytes` on Python 3? If yes, why is value not always the same type?

Comment (tvalentyn) [reply sdks/python/apache_beam/io/tfrecordio.py]: +1. I want us to have clarity on which types we expect throughout methods of TFRecord IO, we should avoid having  APIs that accept both `str=unicode` and `bytes`. See: https://docs.python.org/3/howto/pyporting.html#text-versus-binary-data. If it makes sense for TF Records to hold unencoded strings, then we may want to encode them to bytes right before writing them, and decode them as soon as we read them.

File: sdks/python/apache_beam/io/tfrecordio.py [hunk 2]
```diff
@@ -96,11 +99,12 @@ def write_record(cls, file_handle, value):
       value: A string content of the record.
     """
     encoded_length = struct.pack('<Q', len(value))
```
Comment (tvalentyn) [inline sdks/python/apache_beam/io/tfrecordio.py]: I think we need to take a closer look at this IO regarding `str` vs `bytes`:
- we may want to compute encoded_length after `value` is encoded to bytes.
- we need to clarify in the docstring what type is returned as a result of `read_record` and what type is expected as argument to `encoded_num_bytes`. 
- we should also have unicode strings in tfrecordio_test.

File: sdks/python/apache_beam/io/tfrecordio.py (hunk 1 above)
Comment (ruoyu90) [reply sdks/python/apache_beam/io/tfrecordio.py]: Thanks for all the reviews 😃 I rolled back my previous change and only make sure the write_record function always operate on bytes for both py2 and py3

**Comment by robertwb** [comment]
Thanks. This now LGTM.

**Pull request merged**